src/compiler/checker.ts

```diff
@@ -10136,13 +10137,37 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         })!.parent;
     }
 
+    function getInstanceTypeOfClassSymbol(classSymbol: Symbol): Type {
+        const classType = getDeclaredTypeOfSymbol(classSymbol) as GenericType;
+        const objectFlags = getObjectFlags(classType);
+        if (!(objectFlags & ObjectFlags.ClassOrInterface) || !classType.typeParameters) {
+            return classType;
+        }
+        const variances = getVariances(classType);
+        const typeArguments = map(classType.typeParameters, (typeParameter, i) => {
+            if (!strictInstanceOfTypeParameters) {
+                return anyType;
+            }
+            const variance = variances[i];
+            switch (variance & VarianceFlags.VarianceMask) {
+                case VarianceFlags.Independent:
+                case VarianceFlags.Bivariant:
+                    return anyType;

```

Comment:
So it seems like we wanna try removing _all_ the `any`s from `instanceof` with this, providing it's not too breaky. So.
`Bivariant` is assigned to contravariant positions in methods, and is superceded by any pure co- or contra- variant uses in the same type. So it's secretly just relaxed `Contravariant`. So it should be `never`. `Independent` could be anything safely, since it's for unreferenced type parameters. While `any` likely would save us the most work, the constraint is likely fine (and probably better for accidentally-invariant parameters which become covariant (the most common variance) as people introduce usages).

So if we could try making `Bivariant` parameters `never` and `Independent` parameters the constraint, we'd like to try that out and run our community code suites with the flag on to see if that's not too breaky. If it's completely non-breaky, we may even drop the flag entirely or bundle it into `noImplicitAny`.

Reply:
Exactly what @weswigham said - #53322 has some sparse details of how we got there.

Reply:
Suggested changes applied! Thanks everyone for the feedback and design decisions, it is all greatly appreciated.